### PR TITLE
spotify-tui: add linux dependencies, use brewed openssl

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/spotify-tui.rb [hunk 1]
```diff
@@ -14,6 +15,12 @@ class SpotifyTui < Formula
   end
 
   depends_on "rust" => :build
+  depends_on "openssl@1.1"
```
Comment (carlocab) [inline Formula/spotify-tui.rb:18]: Did you test whether it finds brewed `openssl` without `pkg-config`?

Comment (carlocab) [reply Formula/spotify-tui.rb:18]: Using the formula in this PR:
```
❯ brew linkage spotify-tui
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libobjc.A.dylib
  /usr/lib/libresolv.9.dylib
```
Is it there also no direct linkage on Linux?

Comment (iMichka) [reply Formula/spotify-tui.rb:18]: On linux:
```
➜  Formula git:(master) brew linkage spotify-tui
System libraries:
  /lib/x86_64-linux-gnu/libc.so.6
  /lib/x86_64-linux-gnu/libdl.so.2
  /lib/x86_64-linux-gnu/libm.so.6
  /lib/x86_64-linux-gnu/libpthread.so.0
  /lib/x86_64-linux-gnu/librt.so.1
  /lib64/ld-linux-x86-64.so.2
Homebrew libraries:
  /home/linuxbrew/.linuxbrew/lib/libgcc_s.so.1 (gcc)
  /home/linuxbrew/.linuxbrew/lib/libxcb-render.so.0 (libxcb)
  /home/linuxbrew/.linuxbrew/lib/libxcb-shape.so.0 (libxcb)
  /home/linuxbrew/.linuxbrew/lib/libxcb-xfixes.so.0 (libxcb)
  /home/linuxbrew/.linuxbrew/lib/libxcb.so.1 (libxcb)
  /home/linuxbrew/.linuxbrew/lib/libcrypto.so.1.1 (openssl@1.1)
  /home/linuxbrew/.linuxbrew/lib/libssl.so.1.1 (openssl@1.1)
```

Comment (iMichka) [reply Formula/spotify-tui.rb]: Looks like it does not care at all about openssl on Mac. I added it as a linux-only dependency. It's weird though.

Comment (carlocab) [reply Formula/spotify-tui.rb]: Yea, looks like. I tried building it with both `openssl` and `pkg-config` and there's still nothing.

**Review by carlocab — APPROVED** [review]

**Pull request merged**